File: Data Structures & Algorithms/maximum-product-subarray/submission-2.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    maxProduct(nums) {
        let res = Math.min(...nums);
        let max = 1;
        let min = 1;
        
        for (let num of nums) {
            let temp = max;
            // 1. 找出本輪最大值
            max = Math.max(max * num, min * num, num);
            // 2. 找出本輪最小值
            min = Math.min(temp * num, min * num, num);
            // 3. 使用 res 記錄到目前為止的最大值
            res = Math.max(res, max);
        }

        return res;
    }
}

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        num_set = set(nums)
        longest = 0

        for num in nums:
            # 確認這個數字是這個 subsequence 裡面最小的再開始跑
            if (num-1) not in num_set:
                length = 1
                while (num+length) in num_set:
                    length += 1
                longest = max(longest, length)
        
        return longest